projects/ad77681evb/common/ad77681evb_bd.tcl

```diff
@@ -64,157 +46,55 @@ current_bd_instance /spi_adc1
 
-  ad_connect resetn axi_2/s_axi_aresetn
-  ad_connect irq axi_2/irq
+# dma for the ADC1
 
-current_bd_instance /
+ad_ip_instance axi_dmac axi_ad77681_dma
+ad_ip_parameter axi_ad77681_dma CONFIG.DMA_TYPE_SRC 1
+ad_ip_parameter axi_ad77681_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_ad77681_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_ad77681_dma CONFIG.SYNC_TRANSFER_START 0
+ad_ip_parameter axi_ad77681_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter axi_ad77681_dma CONFIG.AXI_SLICE_DEST 1
+ad_ip_parameter axi_ad77681_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_ad77681_dma CONFIG.DMA_DATA_WIDTH_SRC 32
+ad_ip_parameter axi_ad77681_dma CONFIG.DMA_DATA_WIDTH_DEST 64
 
-ad_connect adc1_data_ready spi_adc1/drdy
-ad_connect adc2_data_ready spi_adc2/drdy
+ad_connect  sys_cpu_clk spi_adc/clk
+ad_connect  sys_cpu_resetn spi_adc/resetn
+ad_connect  sys_cpu_resetn axi_ad77681_dma/m_dest_axi_aresetn
 
-# dma for the ADC1
+ad_connect  spi_adc/m_spi adc_spi
+ad_connect  axi_ad77681_dma/s_axis spi_adc/M_AXIS_SAMPLE
 
-ad_ip_instance axi_dmac axi_ad77681_dma_1
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.DMA_TYPE_SRC 1
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.DMA_TYPE_DEST 0
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.CYCLIC 0
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.SYNC_TRANSFER_START 0
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.AXI_SLICE_SRC 0
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.AXI_SLICE_DEST 1
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.DMA_2D_TRANSFER 0
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.DMA_DATA_WIDTH_SRC 32
-ad_ip_parameter axi_ad77681_dma_1 CONFIG.DMA_DATA_WIDTH_DEST 64
-
-ad_connect  sys_cpu_clk spi_adc1/clk
-ad_connect  sys_cpu_resetn spi_adc1/resetn
-ad_connect  sys_cpu_resetn axi_ad77681_dma_1/m_dest_axi_aresetn
-
-ad_connect  spi_adc1/m_spi adc1_spi
-ad_connect  axi_ad77681_dma_1/s_axis spi_adc1/M_AXIS_SAMPLE
-
-# dma for the ADC2
-
-ad_ip_instance axi_dmac axi_ad77681_dma_2
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.DMA_TYPE_SRC 1
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.DMA_TYPE_DEST 0
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.CYCLIC 0
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.SYNC_TRANSFER_START 0
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.AXI_SLICE_SRC 0
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.AXI_SLICE_DEST 1
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.DMA_2D_TRANSFER 0
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.DMA_DATA_WIDTH_SRC 32
-ad_ip_parameter axi_ad77681_dma_2 CONFIG.DMA_DATA_WIDTH_DEST 64
-
-ad_connect  sys_cpu_clk spi_adc2/clk
-ad_connect  sys_cpu_resetn spi_adc2/resetn
-ad_connect  sys_cpu_resetn axi_ad77681_dma_2/m_dest_axi_aresetn
-
-ad_connect  spi_adc2/m_spi adc2_spi
-ad_connect  axi_ad77681_dma_2/s_axis spi_adc2/M_AXIS_SAMPLE
 
 # AXI address definitions
 
-ad_cpu_interconnect 0x44a00000 spi_adc1/axi_1
-ad_cpu_interconnect 0x44b00000 spi_adc2/axi_2
-ad_cpu_interconnect 0x44a30000 axi_ad77681_dma_1
-ad_cpu_interconnect 0x44b30000 axi_ad77681_dma_2
+ad_cpu_interconnect 0x44a00000 spi_adc/axi_1
```

Comment:
Let's change the "axi_1" to something else.  Some suggestions are axi_regmap or axi_interface.